File: module_domain/src/main/java/project/redis/movie/MovieGenre.java
package project.redis.movie;

import lombok.Getter;

@Getter
public enum MovieGenre {
    ACTION("액션"),
    ROMANCE("로맨스"),
    HORROR("호러"),
    SF("SF");

    private final String movieGenreDescription;

    MovieGenre(String movieGenreDescription) {
        this.movieGenreDescription = movieGenreDescription;
    }
}

File: domain/src/main/kotlin/com/hanghe/redis/movie/MovieGenre.kt
package com.hanghe.redis.movie

enum class MovieGenre(val type: String) {
    ACTION("액션"),

    ROMANCE("로맨스"),

    HORROR("호러"),

    SF("SF");

    companion object {
        fun from(genre: String): MovieGenre? {
            return entries.find { it.name == genre }
        }
    }
}
